pkg/user/signer.go

```diff
@@ -0,0 +1,375 @@
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := s.BroadcastTx(ctx, txBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.ConfirmTx(ctx, resp.TxHash)
+}
+
+// SubmitPayForBlob forms a transaction from the provided blobs, signs it, and submits it to the chain.
+// TxOptions may be provided to set the fee and gas limit.
+func (s *Signer) SubmitPayForBlob(ctx context.Context, blobs []*tmproto.Blob, opts ...TxOption) (*sdktypes.TxResponse, error) {
+	txBytes, err := s.CreatePayForBlob(blobs, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := s.BroadcastTx(ctx, txBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.ConfirmTx(ctx, resp.TxHash)
+}
+
+// CreateTx forms a transaction from the provided messages and signs it. TxOptions may be optionally
+// used to set the gas limit and fee.
+func (s *Signer) CreateTx(msgs []sdktypes.Msg, opts ...TxOption) ([]byte, error) {
+	txBuilder := s.txBuilder(opts...)
+	if err := txBuilder.SetMsgs(msgs...); err != nil {
+		return nil, err
+	}
+
+	if err := s.signTransaction(txBuilder); err != nil {
+		return nil, err
+	}
+
+	return s.encCfg.TxConfig.TxEncoder()(txBuilder.GetTx())
+}
+
+func (s *Signer) CreatePayForBlob(blobs []*tmproto.Blob, opts ...TxOption) ([]byte, error) {
+	msg, err := blob.NewMsgPayForBlobs(s.address.String(), blobs...)
+	if err != nil {
+		return nil, err
+	}
+
+	txBytes, err := s.CreateTx([]sdktypes.Msg{msg}, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return tmtypes.MarshalBlobTx(txBytes, blobs...)
+}
+
+// BroadcastTx submits the provided transaction bytes to the chain and returns the response.
+func (s *Signer) BroadcastTx(ctx context.Context, txBytes []byte) (*sdktypes.TxResponse, error) {
+	txClient := tx.NewServiceClient(s.grpc)
+
+	// TODO (@cmwaters): handle nonce mismatch errors
+	resp, err := txClient.BroadcastTx(
+		ctx,
+		&tx.BroadcastTxRequest{
+			Mode:    tx.BroadcastMode_BROADCAST_MODE_SYNC,
+			TxBytes: txBytes,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return resp.TxResponse, nil
+}
+
+// ConfirmTx periodically pings the provided node for the commitment of a transaction by its
+// hash. It will continually loop until the context is cancelled, the tx is found or an error
+// is encountered.
+func (s *Signer) ConfirmTx(ctx context.Context, txHash string) (*sdktypes.TxResponse, error) {
+	txClient := tx.NewServiceClient(s.grpc)
+
+	resp, err := txClient.GetTx(
+		ctx,
+		&tx.GetTxRequest{
+			Hash: txHash,
+		},
+	)
+	if err == nil {
+		return resp.TxResponse, err
+	}
+
+	// this is a bit brittle
+	if !strings.Contains(err.Error(), "not found") {
+		return &sdktypes.TxResponse{}, err
+	}
+
+	timer := time.NewTimer(s.pollTime)
+	for {
+		select {
+		case <-ctx.Done():
+			return &sdktypes.TxResponse{}, ctx.Err()
+		case <-timer.C:
+			resp, err = txClient.GetTx(
+				ctx,
+				&tx.GetTxRequest{
+					Hash: txHash,
+				},
+			)
+
+			if err == nil {
+				return resp.TxResponse, err
```

Comment:
[nit] For improved readability
```suggestion
				return resp.TxResponse, nil
```